src/bootstrap/lib.rs

```diff
@@ -1431,6 +1431,13 @@ impl Build {
             return Vec::new();
         }
 
+        if !stamp.exists() {
+            panic!(
+                "Could not find stamp file {}, did you try to keep a nonexistent build stage?",
+                stamp.display()
```

Comment:
This could use eprintln and detail_exit instead.

Reply:
I could only remember println and process::exit, if eprintln and detail_exit is the better option, I'll change it when I get home

Reply:
`detail_exit` is a custom function located in the root of bootstrap (`lib.rs`), and `eprintln` is a macro like `println`, but prints to stderr instead of stdout.
Also, do not create merge commits, rather [rebase](https://rustc-dev-guide.rust-lang.org/git.html#rebasing) when conflicts occur.

Reply:
My mistake, and it definitely seems like I didn't dig deep enough into ways to error and quit.

Reply:
no worries :) every codebase has its own custom errors, not a big deal to learn it as you go